src/core/net/ip6_address.hpp

```diff
@@ -324,6 +324,18 @@ class Address : public otIp6Address, public Equatable<Address>, public Clearable
         kGlobalScope         = 14, ///< Global scope
     };
 
+    /**
+     * This enumeration defines IPv6 address type filter.
+     *
+     */
+    enum TypeFilter : uint8_t
```

Comment:
```suggestion
    enum Filter : uint8_t
```
nit: Filters `kTypeMulticast` and `kTypeMulticastLargerThanRealmLocal` are filtering addresses of same type (Multicast) but with different scope range? Maybe rename to `Filter` to make it more general?

Reply:
I prefer to keep it `TypeFilter` to be clear that it is a "type filter"  (i.e. filtering based on address type) and avoid confusion with address filtering and other definition (e.g., we have a `Ip6::Filter` class).

Reply:
Fine with me.